yarn-project/aztec.js/src/fee/fee_payment_method.ts

```diff
@@ -6,19 +6,17 @@ import { type AztecAddress } from '@aztec/foundation/aztec-address';
  * Holds information about how the fee for a transaction is to be paid.
  */
 export interface FeePaymentMethod {
-  /**
-   * The asset used to pay the fee.
-   */
+  /** The asset used to pay the fee. */
   getAsset(): AztecAddress;
-  /**
-   * Address which will hold the fee payment.
-   */
-  getPaymentContract(): AztecAddress;
-
   /**
    * Creates a function call to pay the fee in the given asset.
    * @param gasSettings - The gas limits and max fees.
    * @returns The function call to pay the fee.
    */
   getFunctionCalls(gasSettings: GasSettings): Promise<FunctionCall[]>;
+  /**
+   * Whether the sender should be appointed as fee payer.
+   * @param gasSettings - The gas limits and max fees.
+   */
+  isFeePayer(gasSettings: GasSettings): Promise<boolean>;
```

Comment:
Seems a bit weird for a "payment method" to act as a fee payer?

Reply:
Maybe this could be `getFeePayer(): AztecAddress` and have the entrypoint check if `context.this_address() == fee_payer_address` instead of a boolean?

Reply:
Good point. I'll change it but make the comparison in the entrypoint in ts-land, if you think it's ok.

Reply:
Fixed!